pandas/core/generic.py

```diff
@@ -3151,150 +3167,338 @@ def to_latex(
-            ``\begin{{}}`` in the output.
+            .. deprecated:: 1.4.0
+               Adding LaTeX styling commands renders spacing not applicable.
+        bold_rows : bool, default False
+            Unused.
 
-            .. versionadded:: 1.2.0
+            .. deprecated:: 1.4.0
+               Use ``bold_header`` instead.
         {returns}
         See Also
         --------
+        Styler.to_latex : Render a Styler to LaTeX.
         DataFrame.to_string : Render a DataFrame to a console-friendly
             tabular output.
         DataFrame.to_html : Render a DataFrame as an HTML table.
 
+        Notes
+        -----
+
+        .. note::
+           As of version 1.4.0 this method was changed to use the implementation of
+           `Styler.to_latex()` via jinja2 templating, and no longer uses the pandas
+           internal `LatexFormatter` class. This is the reason for many of the
+           altered arguments in the method signature.
+
+           The Styler implementation has extended functionality and performance and
+           further information is available in its own documentation relevant for this
+           method.
+
         Examples
         --------
-        >>> df = pd.DataFrame(dict(name=['Raphael', 'Donatello'],
-        ...                   mask=['red', 'purple'],
-        ...                   weapon=['sai', 'bo staff']))
-        >>> print(df.to_latex(index=False))  # doctest: +NORMALIZE_WHITESPACE
+        >>> df = pd.DataFrame({{'name': ['Raphael', 'Donatello'],
+        ...                    'mask': ['red', 'purple'],
+        ...                    'weapon': ['sai', 'bo staff']}})
+        >>> print(df.to_latex(index=False, hrules=True))
         \begin{{tabular}}{{lll}}
-         \toprule
-               name &    mask &    weapon \\
-         \midrule
-            Raphael &     red &       sai \\
-          Donatello &  purple &  bo staff \\
+        \toprule
+        name & mask & weapon \\
+        \midrule
+        Raphael & red & sai \\
+        Donatello & purple & bo staff \\
         \bottomrule
         \end{{tabular}}
         """
-        # Get defaults from the pandas config
-        if self.ndim == 1:
-            self = self.to_frame()
-        if longtable is None:
-            longtable = config.get_option("display.latex.longtable")
-        if escape is None:
-            escape = config.get_option("display.latex.escape")
-        if multicolumn is None:
-            multicolumn = config.get_option("display.latex.multicolumn")
-        if multicolumn_format is None:
-            multicolumn_format = config.get_option("display.latex.multicolumn_format")
-        if multirow is None:
-            multirow = config.get_option("display.latex.multirow")
-
-        self = cast("DataFrame", self)
-        formatter = DataFrameFormatter(
-            self,
-            columns=columns,
-            col_space=col_space,
+        kwargs = locals()
```

Comment:
i don't want to do any of this here. better in io/format/latext.py (or you can make a latex_deprecated or similar).